cmd/ddev/cmd/debug-rebuild.go

```diff
@@ -54,44 +54,87 @@ var DebugRebuildCmd = &cobra.Command{
 		}
 
 		app.DockerEnv()
+
 		if err = app.WriteDockerComposeYAML(); err != nil {
 			util.Failed("Failed to get compose-config: %v", err)
 		}
 
-		output.UserOut.Printf("Rebuilding project images...")
 		buildDurationStart := util.ElapsedDuration(time.Now())
 		composeRenderedPath := app.DockerComposeFullRenderedYAMLPath()
 		withoutCache := !cmd.Flags().Changed("cache")
 
 		buildArgs := []string{"-f", composeRenderedPath, "--progress", "plain", "build"}
 
 		if !buildAll {
-			buildArgs = append(buildArgs, cmd.Flag("service").Value.String())
+			buildArgs = append(buildArgs, service)
 		}
 
 		if withoutCache {
 			buildArgs = append(buildArgs, "--no-cache")
+			output.UserOut.Printf("Rebuilding project images without Docker cache...")
+		} else {
+			output.UserOut.Printf("Rebuilding project images using Docker cache...")
 		}
 
-		util.Success("Rebuilding project %s with `%s %v`", app.Name, composeBinaryPath, strings.Join(buildArgs, " "))
-
+		output.UserOut.Printf("Executing `%s %v`", composeBinaryPath, prettyCmd(buildArgs))
 		err = exec2.RunInteractiveCommand(composeBinaryPath, buildArgs)
 		if err != nil {
-			util.Failed("Failed to execute `%s %v`: %v", composeBinaryPath, strings.Join(buildArgs, " "), err)
+			util.Failed("Failed to execute `%s %v`: %v", composeBinaryPath, prettyCmd(buildArgs), err)
 		}
+
 		buildDuration := util.FormatDuration(buildDurationStart())
-		util.Success("Rebuilt Docker cache for project %s in %s", app.Name, buildDuration)
+		if buildAll {
+			util.Success("Rebuilt %s cache in %s", app.Name, buildDuration)
+		} else {
+			util.Success("Rebuilt %s service cache for %s in %s", service, app.Name, buildDuration)
+		}
 
-		err = app.Restart()
-		if err != nil {
-			util.Failed("Failed to restart project: %v", err)
+		// Restart the entire project only when changing "web",
+		// since app.Start() includes a lot of extra logic
+		// and just restarting the web service isn't enough here
+		if buildAll || service == "web" {
+			err = app.Restart()
+			if err != nil {
+				util.Failed("Failed to restart project: %v", err)
+			}
+			util.Success("Restarted %s", app.GetName())
+			return
+		}
+
+		labels := map[string]string{
+			"com.ddev.site-name":         app.GetName(),
+			"com.docker.compose.oneoff":  "False",
+			"com.docker.compose.service": service,
+		}
+
+		// Restart the specified service using docker-compose, if it is running
+		if container, err := dockerutil.FindContainerByLabels(labels); err == nil && container != nil {
```

Comment:
Note: I changed the logic here to restart the service only if it's already running, since the command is called "rebuild". There's no reason to start it if the project isn't running.

I kept the restart logic for the default case with no arguments, so it's backward compatible with the original `ddev debug rebuild` (`ddev debug refresh`).